web-common/src/features/entity-management/file-artifacts-store.ts

```diff
@@ -1,131 +1,241 @@
-import type { V1ReconcileError } from "@rilldata/web-common/runtime-client";
-import { derived, Readable, writable } from "svelte/store";
-import { parseDocument } from "yaml";
-import type { MetricsConfig } from "../metrics-views/metrics-internal-store";
-
-export type FileArtifactsData = {
-  // name used for the file. will not always be dependent on the file
-  name?: string;
-  errors?: Array<V1ReconcileError>;
-  jsonRepresentation?: MetricsConfig | Record<string, never>;
-  isReconciling?: boolean;
-};
-
-/**
- * Store to save data for each file artifact.
- * Currently, stores reconcile errors
- */
-export type FileArtifactsState = {
-  entities: Record<string, FileArtifactsData>;
-};
-const { update, subscribe } = writable({
-  entities: {},
-} as FileArtifactsState);
-
-const createOrUpdateFileArtifact = (
-  path: string,
-  callback: (entityData: FileArtifactsData) => void,
-) => {
-  update((state) => {
-    if (!state.entities[path]) {
-      state.entities[path] = {
-        errors: [],
-        jsonRepresentation: {},
-        isReconciling: false,
-      };
+import { removeLeadingSlash } from "@rilldata/web-common/features/entity-management/entity-mappers";
+import {
+  ResourceKind,
+  useProjectParser,
+  useResource,
+} from "@rilldata/web-common/features/entity-management/resource-selectors";
+import { getSubStore } from "@rilldata/web-common/lib/getSubStore";
+import {
+  runtimeServiceListResources,
+  type V1ParseError,
+  V1ReconcileStatus,
+  type V1Resource,
+  type V1ResourceName,
+} from "@rilldata/web-common/runtime-client";
+import type { QueryClient } from "@tanstack/svelte-query";
+import { derived, get, type Readable, writable } from "svelte/store";
+
+export class FileArtifact {
+  public readonly path: string;
+
+  public readonly name = writable<V1ResourceName | undefined>(undefined);
+
+  /**
+   * Last time the state of the resource `kind/name` was updated
+   * used to make sure we do not have unnecessary refreshes
+   */
+  public lastStateUpdatedOn: string | undefined;
```

Comment:
The point of this is to avoid triggering a query to resource. This is set in the resources listener. Writable is not needed since it is not used as a reactive data.

Reply:
So would this be an accurate explanation/comment: "We save this data here from the `WatchResources` response to avoid an extra call to `GetResource`"? 

Though we could do better by using `queryClient.setQueryData()` in the `WatchResources` listener, and accessing this state through the TanStack Query cache for the `GetResource` API, right? And the same goes for the `reconciling` store?

Reply:
Yes, I can add more info stating that it saves data from `WatchResources`.

That would be ideal, but i was facing some issues with consistency. Here is a PR where i tried it: https://github.com/rilldata/rill/pull/4379 . As I mentioned yesterday, we should revisit this once asset explorer is finished.

Reply:
Sweet, makes sense. Sounds good.